vms/volunteer/views.py

```diff
@@ -1,162 +1,182 @@
+'''
+
+class ProfileView(LoginRequiredMixin, DetailView):
+    template_name = 'volunteer/profile.html'
+
+    def get_object(self, queryset=None):
+        volunteer_id = self.kwargs['volunteer_id']
+        obj = Volunteer.objects.get(id=self.kwargs['volunteer_id'])
+        if obj:
+            return obj
+        else:
+            return HttpResponse(status=403)
+
+'''
+  The view generate Report.
+  GenerateReportView calls two other views(ShowFormView, ShowReportListView) within it.
+'''
+
+class GenerateReportView(LoginRequiredMixin, View):
+
+    def get(self, request, *args, **kwargs):
+        view = ShowFormView.as_view()
+        return view(request, *args,**kwargs)
+
+    def post(self, request, *args, **kwargs):
+        view = ShowReportListView.as_view()
+        return view(request, *args, **kwargs)
+
+class ShowFormView(LoginRequiredMixin, FormView):
+    """
+    Displays the form
+    """
+    model = Volunteer
+    form_class = ReportForm
+    template_name = "volunteer/report.html"
+
+
+
+class ShowReportListView(LoginRequiredMixin, ListView):
+    """
+    Generate the report using ListView
+    """
+    template_name = "volunteer/report.html"
+
+    def post(self, request, *args, **kwargs):
+        volunteer_id = self.kwargs['volunteer_id']
+        event_list = get_signed_up_events_for_volunteer(volunteer_id)
+        job_list = get_signed_up_jobs_for_volunteer(volunteer_id)
+        event_name = self.request.POST['event_name']
+        job_name = self.request.POST['job_name']
+        start_date = self.request.POST['start_date']
+        end_date = self.request.POST['end_date']
+        report_list = get_volunteer_report(volunteer_id, event_name, job_name, start_date, end_date)
+        total_hours = calculate_total_report_hours(report_list)
+        return render(request, 'volunteer/report.html',
+                      {'report_list': report_list, 'total_hours': total_hours, 'notification': True,
+                       'job_list': job_list, 'event_list': event_list, 'selected_event': event_name,
+                       'selected_job': job_name})
+@login_required
+def search(request):
+    if request.method == 'POST':
+        form = SearchVolunteerForm(request.POST)
+        if form.is_valid():
+
+            first_name = form.cleaned_data['first_name']
+            last_name = form.cleaned_data['last_name']
+            city = form.cleaned_data['city']
+            state = form.cleaned_data['state']
+            country = form.cleaned_data['country']
+            organization = form.cleaned_data['organization']
+
+            search_result_list = search_volunteers(first_name, last_name, city, state, country, organization)
+            return render(request, 'volunteer/search.html', {'form' : form, 'has_searched' : True, 'search_result_list' : search_result_list})
+    else:
+        form = SearchVolunteerForm()
+
+    return render(request, 'volunteer/search.html', {'form' : form, 'has_searched' : False})
```

Comment:
@amruthasangeeth Complete file is coming in the diff. Please fix it, so that I can review. Only the relevant diff should be present.